python/setup.py

```diff
@@ -402,7 +402,7 @@ def has_ext_modules(foo):
     },
     use_scm_version={"root": "..", "relative_to": __file__},
     setup_requires=['setuptools_scm', 'cython >= 0.23'],
-    install_requires=['numpy >= 1.10', 'six >= 1.0.0'],
+    install_requires=['numpy >= 1.10', 'six >= 1.0.0', 'cloudpickle'],
```

Comment:
Why wasn't this change needed before? Also should this be in 'install_requires' or 'tests_require'?

Reply:
This is what I'm refering to in the PR message above. I tried various ways to install it that doesn't make it a runtime requirement (in particular I tried tests_requires, I think in that case the manylinux1 wheels test failed). Maybe @xhochy has an idea.

It wasn't required before, because none of the tests tried to pickle a function (and compat.py falls back to pickle if cloudpickle is not available).